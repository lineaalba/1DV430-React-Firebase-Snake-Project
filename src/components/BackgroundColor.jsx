/**
 * @author Filippa Jakobsson
 * @version 1.0
 */

import React from 'react'
import Button from 'react-bootstrap/Button'

/**
* Function to change background color of game area.
*/
const BackgroundColor = () => {
    const gameArea = document.querySelector('.game__area')
    const bgColor = document.querySelectorAll('.bg__color')

    for (let i = 0; i < bgColor.length; i++) {
        let color
        bgColor[i].addEventListener('click', e => {

            switch (bgColor[i].textContent) {
                case 'Black':
                    color = '#151719'
                    break
                case 'White':
                    color = 'white'
                    break
                case 'Grey':
                    color = 'grey'
                    break
                default:
                    color = '#151719'
            }

            gameArea.style.background = color 
        })
    }
 
    return (
        <div className="dropdown">
            <Button data-testid="backgroundcolor" className="change__background" variant="outline-dark">Change background</Button>
            <div onClick={BackgroundColor} className="dropdown__content">
                <p data-testid="black" className="bg__color">Black</p>
                <p data-testid="white" className="bg__color">White</p>
                <p data-testid="grey" className="bg__color">Grey</p>
            </div>
        </div> 
    )
}

export default BackgroundColor
