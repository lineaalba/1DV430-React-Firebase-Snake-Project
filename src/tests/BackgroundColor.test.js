import React from 'react'
import { shallow } from 'enzyme'
import BackgroundColor from '../components/BackgroundColor'

describe('Tests the BackgroundColor component', () => {

  it('matches snapshot', () => {
    const wrapper = shallow(<BackgroundColor />)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
