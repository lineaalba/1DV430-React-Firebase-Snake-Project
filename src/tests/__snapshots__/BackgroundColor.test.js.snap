// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the BackgroundColor component matches snapshot 1`] = `"<div class=\\"dropdown\\"><button data-testid=\\"backgroundcolor\\" type=\\"button\\" class=\\"change__background btn btn-outline-dark\\">Change background</button><div class=\\"dropdown__content\\"><p class=\\"bg__color\\">Black</p><p class=\\"bg__color\\">White</p><p class=\\"bg__color\\">Grey</p></div></div>"`;
